### ISSUE TITLE
A question about Metropolis-Hastings sampling

### ISSUE THREAD

**Issue by fancyerii** [vsmolyakov/ml_algo_in_depth#3]
I have a question about this line of code:

```
            #MH ratio
            self.alpha[i] = self.proposal_pdf(x_new) / self.proposal_pdf(x_curr) #q(x|x')/q(x'|x)
```

proposal_pdf is:

```
    def proposal_pdf(self, x):
        #q(x) = N(x; mu, Sigma)
        return multivariate_normal.pdf(x, self.proposal_mu, self.proposal_sigma)
```
 q(x|x') = proposal_pdf(x_new).

According to the book:

![image](https://github.com/user-attachments/assets/b70f991c-960d-4291-a36a-b14f1fa85363)

it seems q(x|x') = N(x; x', Sigma) = multivariate_normal.pdf(x, x_new, self.proposal_sigma)

**Comment by vsmolyakov** [comment]
Thanks! You are right, I updated the code in the following commit: https://github.com/vsmolyakov/ml_algo_in_depth/commit/15003924761e6d77485add8e64a61be00e34cf4a

It looks like the proposal_pdf was always sampling from the same self.proposal_mu mean, I updated it so that the mean is a parameter, in other words:
`q(x|x') = N(x; x', Sigma) = multivariate_normal.pdf(x_curr, x_new, self.proposal_sigma)
`
This resulted in a slightly higher MH acceptance ratio: 0.1674 (new) compared to 0.1586 (old).
Thanks!
